api/rest/bundle_handler.go

```diff
@@ -69,7 +77,20 @@ type BundleHandler struct {
 	clock                 Clock
 	workDir               string                // location where bundles are generated and stored
 	collectors            []collector.Collector // information what should be in the bundle
+	coordinator           Coordinator           // coordinator for building remote bundles
 	bundleCreationTimeout time.Duration         // limits how long bundle creation could take
+	urlBuilder            dcos.NodeURLBuilder
+}
+
+type createArguments struct {
+	BundleType string `json:"type"`
+	Nodes      []node `json:"nodes"`
```

Comment:
How should this be handled by this API? Should we only specify a body if we want a remote bundle and then use `bundleType` to determine if it's `masters`, `all`, or `agents`? Would it make more sense to have two separate endpoints for remote and local in that case?

Reply:
I'm not sure how to handle it properly. I think we should stick with a single endpoint for remote and local. I think we can accept `nodes` as a list of strings. Then we can detect their roles and IPs.

Examples:
- `["leader"]` - only leader (do we need it?)
- `["masters"]` - only masters (the same works for `nodes`
- `["masters", "nodes"]` = `["all"`]` - whole cluster
- `["masters", "192.168.0.1"]` - returns only masters and one additional node (we need to check if selected node was not included in `masters` to prevent creating bundle twice
- `["masters", "192.168.0.1", "node1", "be6587c6-4838-4a4b-904a-d41c5ed9d501"]` 

@drewvanstone @makkes What do you think?